tests/profiler/bob_user_workflow_verbose_profiler_config.yml

```diff
@@ -0,0 +1,216 @@
+        # TODO: Streaming mean / stdev consider how would that work?
+        class_name: MultiBatchAggregateFunctionParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_metric_parameter_builder
+        batch_request:
+          # Filter for batches for previous full year
+          #  using batch_filter_parameters with `year` as one of the parameters
+          #  `year` needs to be defined as a group by the data_connector
+          # TODO: is this possible - how do we specify a data_connector (when calling Profiler.profile()?)?
+          data_connector_query:
+            batch_filter_parameters:
+              year: 2020 # TODO: Can this be relative to the current year? (e.g. specify last year vs the exact year)
+        metric_configuration:
+          metric_name: column.mean # per batch metric
+          # Domain kwargs here refer to the current domain being processed - the list of domains coming from the DomainBuilder (here the list of columns filtered by SimpleColumnSuffixDomainBuilder)
+          metric_domain_kwargs: $domain.domain_kwargs
+        aggregate_fn: mean # Not necessarily a metric, but some function to apply to all of the batches - can be different from metric_name
+      - name: my_stdev_of_means_parameter_for_batches_in_previous_full_year
+        # Parameter computes the stdev of means of all batches in the previous full year
+        class_name: MultiBatchAggregateFunctionParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_metric_parameter_builder
+        batch_request:
+          data_connector_query:
+            batch_filter_parameters:
+              year: 2020
+        metric_configuration:
+          metric_name: column.mean # per batch metric
+          metric_domain_kwargs: $domain.domain_kwargs
+        aggregate_fn: stdev # note different than metric computed in metric_configuration
+    expectation_configuration_builders:
+      - expectation_type: expect_column_values_to_be_between
+        class_name: DefaultExpectationConfigurationBuilder
+        module_name: great_expectations.profiler.expectation_configuration_builder.default_expectation_configuration_builder
+        # Note: literals are and will continue to be supported (you do not have to add to "variables" first, unless you want to reference in multiple locations):
+        min_value: 0
+        max_value: $variables.max_amount
+        column: $domain.domain_kwargs.column
+      - expectation_type: expect_column_mean_to_be_between
+        class_name: DefaultExpectationConfigurationBuilder
+        module_name: great_expectations.profiler.expectation_configuration_builder.default_expectation_configuration_builder
+        column: $domain.domain_kwargs.column
+        # Note: proposal to reference here in dot notation by the aggregate_fn name
+        min_value: $parameter.my_mean_parameter_for_batches_in_previous_full_year.mean * 0.8
```

Comment:
Does the aggregate_fn at the end need to be declared above, or could it be calculated lazily here? For instance, could someone NOT specify an aggregate_fn above, and then reference `$parameter.my_mean_parameter_for_batches_in_previous_full_year.mean` or `$parameter.my_mean_parameter_for_batches_in_previous_full_year.stdev` or `$parameter.my_mean_parameter_for_batches_in_previous_full_year.sum`? Or is the earlier caluclation necessary?

Reply:
Hmm - great question. The way we are currently calculating the parameters is as they are defined so we would need to specify in the parameter builder. But this sounds like a great optimization step, e.g. when we talked about building the computation graph first after processing the whole config before calculating any metrics (also holding off on any metrics hidden behind conditionals until we evaluate the conditional to avoid calculating unnecessary metrics.).

Reply:
Makes sense (though I would propose to defer this optimization till the time we can measure the actual usage patterns).